.github/workflows/autotest.yml

```diff
@@ -5,7 +5,25 @@ on:
   - push
 
 jobs:
+  header-lint:
+    if: 'github.event_name == ''pull_request'''
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: |
+          git pull origin master
```

Comment:
I guess this should not be required since you use `actions/checkout@v2` with a `fetch-depth: 0`. Could you double check this? In the containers repo, I was able to get the revision for the remotes/origin/master as follows, without the need to do a git pull:

https://github.com/rucio/containers/blob/b09dbd05b57ae3dc5774e09fa00dfb79529a6d9b/.github/workflows/docker-auto-build.yml#L102-L103


```
GIT_MASTER_LATEST_COMMIT_SHA=$(echo $(git ls-remote --heads origin | grep master  |  awk -F '\t' '{print $1}'))
```

Reply:
So I'd assume, you would still get the same output from `$(git merge-base HEAD remotes/origin/master)` if you remove the `git pull`

Reply:
We don't need the entire history for the changes, we need it for the `add_header` script.
https://github.com/rucio/rucio/blob/6d0d358e04f5431f0b9a98ae40f31af0ddff4833/tools/add_header#L100

I tried it without the git pull and then the script could not find the current master branch commit sha...

Reply:
Interesting, let's keep it ;)